include/inviwo/core/util/spatialsampler.h

```diff
@@ -37,158 +37,194 @@ namespace inviwo {
-    return withinBounds(static_cast<Vector<SpatialDims, double>>(pos));
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const vec3& pos) const {
+    return withinBounds(static_cast<dvec3>(pos));
 }
-template <unsigned int SpatialDims, unsigned int DataDims, typename T>
-bool SpatialSampler<SpatialDims, DataDims, T>::withinBounds(
-    const Vector<SpatialDims, double>& pos) const {
+
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const vec2& pos) const {
+    return withinBounds(dvec3{pos, 0.0});
+}
+
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const dvec2& pos) const {
+    return withinBounds(dvec3{pos, 0.0});
+}
+
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const dvec3& pos) const {
     if (space_ != Space::Data) {
-        const auto p = transform_ * Vector<SpatialDims + 1, double>(pos, 1.0);
-        return withinBoundsDataSpace(Vector<SpatialDims, double>(p) / p[SpatialDims]);
+        const auto p = transform_ * dvec4(pos, 1.0);
+        return withinBoundsDataSpace(dvec3(p) / p.w);
     } else {
         return withinBoundsDataSpace(pos);
     }
 }
 
-template <unsigned int SpatialDims, unsigned int DataDims, typename T>
-bool SpatialSampler<SpatialDims, DataDims, T>::withinBounds(const Vector<SpatialDims, float>& pos,
-                                                            Space space) const {
-    return withinBounds(static_cast<Vector<SpatialDims, double>>(pos), space);
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const vec3& pos, Space space) const {
+    return withinBounds(static_cast<dvec3>(pos), space);
+}
+
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const vec2& pos, Space space) const {
+    return withinBounds(dvec3{pos, 0.0}, space);
+}
+
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const dvec2& pos, Space space) const {
+    return withinBounds(dvec3{pos, 0.0}, space);
 }
 
-template <unsigned int SpatialDims, unsigned int DataDims, typename T>
-bool SpatialSampler<SpatialDims, DataDims, T>::withinBounds(const Vector<SpatialDims, double>& pos,
-                                                            Space space) const {
+template <unsigned int DataDims, typename T>
+bool SpatialSampler<DataDims, T>::withinBounds(const dvec3& pos, Space space) const {
     if (space != Space::Data) {
-        const Matrix<SpatialDims + 1, double> m{
-            spatialEntity_.getCoordinateTransformer().getMatrix(space, Space::Data)};
-        const auto p = m * Vector<SpatialDims + 1, double>(pos, 1.0);
-        return withinBoundsDataSpace(Vector<SpatialDims, double>(p) / p[SpatialDims]);
+        const dmat4 m{spatialEntity_.getCoordinateTransformer().getMatrix(space, Space::Data)};
```

Comment:
This code seems to be in many places. 
Could have a helper function 
```
dvec3 transform(pos, fromSpace, toSpace, transformer) {
    if (fromSpace != toSpace) {
        const dmat4 m{transformer.getCoordinateTransformer().getMatrix(fromSpace, toSpace)}
        const auto p = m * dvec4(pos, 1.0);
        return (dvec3(p) / p.w
    } else {
        return pos;
    }
}
}
```

Reply:
Not so sure. We want to refactor this differently and not have to figure out this on every sample point but rather when creating the sampler. But Im not sure if that is possible with the current implementation

Reply:
But it might be nice to have this functionality directly in the CoordinateTransformer